backend/service/topology/querybuilder_test.go

```diff
@@ -0,0 +1,233 @@
+				Search: &topologyv1.SearchTopologyRequest_Filter_Search{
+					Field: "metadata.search.field",
+					Text:  "cat",
+				},
+				TypeUrl: "type.googleapis.com/clutch.aws.ec2.v1.AutoscalingGroup",
+				Metadata: map[string]string{
+					"label": "value",
+				},
+			},
+			sort: &topologyv1.SearchTopologyRequest_Sort{
+				Field:     "metadata.meow.iam.a.cat",
+				Direction: topologyv1.SearchTopologyRequest_Sort_ASCENDING,
+			},
+			pageToken: 10,
+			limit:     5,
+			expect:    "SELECT id, data, metadata FROM topology_cache WHERE metadata->'search'->'field' LIKE $1 AND resolver_type_url = $2 AND metadata->>'label' = $3 ORDER BY metadata->'meow'->'iam'->'a'->'cat' ASC LIMIT 5 OFFSET 50",
+		},
+	}
+	for _, test := range testCases {
+		output, err := paginatedQueryBuilder(test.filter, test.sort, test.pageToken, test.limit)
+		assert.NoError(t, err)
+
+		sql, _, err := output.ToSql()
+		assert.NoError(t, err)
+		assert.Equal(t, test.expect, sql)
+	}
+}
+
+func TestFilterQueryBuilder(t *testing.T) {
+	testCases := []struct {
+		id     string
+		input  *topologyv1.SearchTopologyRequest_Filter
+		expect string
+	}{
+		{
+			id:     "No Input",
+			input:  &topologyv1.SearchTopologyRequest_Filter{},
+			expect: "SELECT * FROM topology_cache",
+		},
+		{
+			id: "Search by column",
+			input: &topologyv1.SearchTopologyRequest_Filter{
+				Search: &topologyv1.SearchTopologyRequest_Filter_Search{
+					Field: "column.id",
+					Text:  "cat",
+				},
+			},
+			expect: "SELECT * FROM topology_cache WHERE id LIKE $1",
+		},
+		{
+			id: "Search by Metadata",
+			input: &topologyv1.SearchTopologyRequest_Filter{
+				Search: &topologyv1.SearchTopologyRequest_Filter_Search{
+					Field: "metadata.label",
+					Text:  "cat",
+				},
+			},
+			expect: "SELECT * FROM topology_cache WHERE metadata->>'label' LIKE $1",
+		},
+		{
+			id: "Search all options",
+			input: &topologyv1.SearchTopologyRequest_Filter{
+				Search: &topologyv1.SearchTopologyRequest_Filter_Search{
+					Field: "metadata.label",
+					Text:  "cat",
+				},
+				TypeUrl: "type.googleapis.com/clutch.aws.ec2.v1.AutoscalingGroup",
+				Metadata: map[string]string{
+					"label":  "value",
+					"label2": "value2",
+				},
+			},
+			expect: "SELECT * FROM topology_cache WHERE metadata->>'label' LIKE $1 AND resolver_type_url = $2 AND metadata->>'label' = $3 AND metadata->>'label2' = $4",
+		},
+	}
+
+	for _, test := range testCases {
+		selectBuilder := sq.StatementBuilder.PlaceholderFormat(sq.Dollar).
+			Select("*").
+			From("topology_cache")
+
+		output := filterQueryBuilder(selectBuilder, test.input)
+		sql, _, err := output.ToSql()
+		assert.NoError(t, err)
+		assert.Equal(t, test.expect, sql)
+	}
+}
+
+func TestSortQueryBuilder(t *testing.T) {
+	testCases := []struct {
+		id     string
+		input  *topologyv1.SearchTopologyRequest_Sort
+		expect string
+	}{
+		{
+			id:     "Default to ID ASC",
+			input:  &topologyv1.SearchTopologyRequest_Sort{},
```

Comment:
if only Direction is specified and not the field, is the field expected to default to ID? or is that not a valid input?

Reply:
currently its a required field https://github.com/lyft/clutch/blob/main/api/topology/v1/topology_api.proto#L56